libs/grok/src/main/java/org/elasticsearch/grok/GrokBuiltinPatterns.java

```diff
@@ -0,0 +1,148 @@
+    /**
+     * Patterns built in to the grok library.
+     */
+    private static Map<String, String> LEGACY_PATTERNS;
+    private static Map<String, String> ECS_V1_PATTERNS;
+
+    public static synchronized Map<String, String> legacyPatterns() {
+        return get(false);
+    }
+
+    public static synchronized Map<String, String> ecsV1Patterns() {
+        return get(true);
+    }
+
+    /**
+     * Load built-in patterns.
+     */
+    public static synchronized Map<String, String> get(boolean ecsCompatibility) {
+        if (ecsCompatibility) {
+            if (ECS_V1_PATTERNS == null) {
+                ECS_V1_PATTERNS = loadEcsPatterns();
+            }
+            return ECS_V1_PATTERNS;
+        } else {
+            if (LEGACY_PATTERNS == null) {
+                LEGACY_PATTERNS = loadLegacyPatterns();
+            }
+            return LEGACY_PATTERNS;
+        }
+    }
+
+    public static Map<String, String> get(String ecsCompatibility) {
+        if (isValidEcsCompatibilityMode(ecsCompatibility)) {
+            return get(ECS_COMPATIBILITY_MODES[1].equals(ecsCompatibility));
+        } else {
+            throw new IllegalArgumentException("unsupported ECS compatibility mode [" + ecsCompatibility + "]");
+        }
+    }
+
+    public static boolean isValidEcsCompatibilityMode(String ecsCompatibility) {
+        return Arrays.asList(ECS_COMPATIBILITY_MODES).contains(ecsCompatibility);
+    }
+
+    private static Map<String, String> loadLegacyPatterns() {
+        String[] patternNames = {
+            "aws",
+            "bacula",
+            "bind",
+            "bro",
+            "exim",
+            "firewalls",
+            "grok-patterns",
+            "haproxy",
+            "httpd",
+            "java",
+            "junos",
+            "linux-syslog",
+            "maven",
+            "mcollective-patterns",
+            "mongodb",
+            "nagios",
+            "postgresql",
+            "rails",
+            "redis",
+            "ruby",
+            "squid" };
+        return readPatternFromDirectory(patternNames, "/patterns/legacy/");
+    }
+
+    private static Map<String, String> loadEcsPatterns() {
+        String[] patternNames = {
+            "aws",
+            "bacula",
+            "bind",
+            "bro",
+            "exim",
+            "firewalls",
+            "grok-patterns",
+            "haproxy",
+            "httpd",
+            "java",
+            "junos",
+            "linux-syslog",
+            "maven",
+            "mcollective",
+            "mongodb",
+            "nagios",
+            "postgresql",
+            "rails",
+            "redis",
+            "ruby",
+            "squid",
+            "zeek" };
+        return readPatternFromDirectory(patternNames, "/patterns/ecs-v1/");
+    }
+
+    private static Map<String, String> readPatternFromDirectory(String[] patternNames, String directory) {
```

Comment:
`s/readPatternFromDirectory/readPatternsFromDirectory/g` (that is, add an 's') since it's receiving many patternNames.

Also I'd probably recommend using `List.of` and `List<String> patternNames`.

Reply:
done, also renamed the method to `loadPatternsFromDirectory` to keep similar names between methods